File: platform/node/test/fixtures/image-agent.mjs
/**
 * Test fixture: echoes the native images it received on the prompt (v2).
 */
import { run } from "../../src/runtime.js";

run({
  name: "image-agent",
  async onPrompt(ctx) {
    ctx.assistantStart();
    const summary = ctx.images
      .map((im) => `${im.mimeType}:${im.data}`)
      .join(";");
    ctx.assistantDelta(`images=${ctx.images.length}|${summary}`);
    ctx.assistantEnd(`images=${ctx.images.length}`);
  },
});

File: platform/python/tests/fixtures/image_agent.py
"""Test fixture: echoes the native images it received on the prompt (v2)."""

from llmtoolforge_agent import run


def on_prompt(ctx):
    ctx.assistant_start()
    summary = ";".join(
        f"{im.get('mimeType')}:{im.get('data')}" for im in ctx.images
    )
    ctx.assistant_delta(f"images={len(ctx.images)}|{summary}")
    ctx.assistant_end(f"images={len(ctx.images)}")


if __name__ == "__main__":
    run(on_prompt, name="image-agent")
